src/parser/type_checking.c

```diff
@@ -24378,7 +24378,7 @@ static PT_TYPE_ENUM
 pt_wrap_type_for_collation (const PT_NODE * arg1, const PT_NODE * arg2, const PT_NODE * arg3,
 			    PT_TYPE_ENUM * wrap_type_collection)
 {
-  PT_TYPE_ENUM common_type = PT_TYPE_CHAR;
+  PT_TYPE_ENUM common_type = (PT_IS_COLLECTION_TYPE (arg1->type_enum))?PT_TYPE_CHAR:PT_TYPE_VARCHAR;
```

Comment:
Why is it a char type only when it is a collection type?

Reply:
This line is changed to PT_TYPE_CHAR from PT_TYPE_VARCHAR at #2483.
It would better that reverting this line to the previous implementation and resolve common_type which starts from type_checking:24425.

Reply:
in #2483, for collection type it's changed to CHAR but host-variable is also set to CHAR, so I modified to set CHAR for collection type only

Reply:
For what test case did you change to CHAR in #2483? In my opinion, it would be better to specify the test case for history management.

Reply:
for #2483
medium/_02_xtests/cases/settyp3.sql
create class set_attr(
se_object set(object),
se_obj_char set( set_attr, string));
insert into set_attr values({},{}) to s2;